watcher_test.go test fails

$ go test github.com/limetext/backend/packages/...
[08/21/16 18:27:12] [DEBG] Scanning testdata for packages
[08/21/16 18:27:12] [DEBG] Scanning testdata for packages
--- FAIL: TestWatchDir (0.10s)
        watcher_test.go:28: Expected package loaded
FAIL
FAIL    github.com/limetext/backend/packages    0.411s

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/37142935-watcher_test-go-test-fails?utm_campaign=plugin&utm_content=tracker%2F15716642&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F15716642&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/packages/watcher_test.go
+++ b/packages/watcher_test.go
@@ -8,11 +8,12 @@ import (
 	"os"
 	"testing"
 	"time"
+	"strings"
 )
 
 func TestWatchDir(t *testing.T) {
 	pkg := &dummyPackage{path: "testdata/file"}
-	rec := &Record{func(s string) bool { return s == "testdata/file" },
+	rec := &Record{func(s string) bool { return strings.Contains(s, "testdata/file") },
 		func(s string) Package { return pkg }}
 
 	Register(rec)

--- a/packages/watcher_test.go
+++ b/packages/watcher_test.go
@@ -8,22 +8,29 @@ import (
 	"os"
 	"testing"
 	"time"
-	"strings"
+	"path/filepath"
 )
 
 func TestWatchDir(t *testing.T) {
-	pkg := &dummyPackage{path: "testdata/file"}
-	rec := &Record{func(s string) bool { return strings.Contains(s, "testdata/file") },
+	// The backend libraries expect absolute paths
+	TestPath, TestPathErr := filepath.Abs("testdata/file")
+
+	if TestPathErr != nil {
+		t.Fatalf("Couldn't get absolute path for testdata/file: %s", TestPathErr)
+	}
+
+	pkg := &dummyPackage{path: TestPath}
+	rec := &Record{func(s string) bool { return s == TestPath },
 		func(s string) Package { return pkg }}
 
 	Register(rec)
 	defer Unregister(rec)
-	watchDir("testdata")
+	watchDir(filepath.Dir(TestPath))
 
-	if _, err := os.Create("testdata/file"); err != nil {
-		t.Fatalf("Error creating 'testdata/file' file: %s", err)
+	if _, err := os.Create(TestPath); err != nil {
+		t.Fatalf("Error creating '%s' file: %s", TestPath, err)
 	}
-	defer os.Remove("testdata/file")
+	defer os.Remove(TestPath)
 	time.Sleep(100 * time.Millisecond)
 	if !pkg.IsLoaded() {
 		t.Error("Expected package loaded")

